### PR TITLE
fixed a permadiff in `google_monitoring_uptime_check_config.http_check.path`

### PR DESCRIPTION
**Pull request by shuyama1**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/10884

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
monitoring: fixed a permadiff when `google_monitoring_uptime_check_config.http_check.path` does not begin with "/"
```

### CODE REVIEW

File: mmv1/templates/terraform/constants/monitoring_uptime_check_config.go.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+func resourceMonitoringUptimeCheckConfigHttpCheckPathDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+	return old == "/"+new
```
Comment (c2thorn) [inline mmv1/templates/terraform/constants/monitoring_uptime_check_config.go.erb:2]: Will the API always add the `/` ?

Comment (shuyama1) [reply mmv1/templates/terraform/constants/monitoring_uptime_check_config.go.erb:2]: Only when the user specified path does not start with "/"

Comment (shuyama1) [reply mmv1/templates/terraform/constants/monitoring_uptime_check_config.go.erb:2]: I think DiffSuppressFunc only works on detected diffs, so if no diffs detected (in this case, if user's config and API return values both start with "/", it should create no diffs) the diffsuppress will not take effect.

We also have other tests with path start with "/", and it seems they all passed the tests https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstreamVcr/268585?buildTab=tests&status=all&name=TestAccMonitoringUptimeCheckConfig&pager.currentPage=1

Comment (c2thorn) [reply mmv1/templates/terraform/constants/monitoring_uptime_check_config.go.erb:2]: That makes a lot of sense, thanks for the clarification!

**Pull request merged**